### PR TITLE
2021-04 Cumulative Quality Rollup for Cuberite

### PR DESCRIPTION
**Pull request by tigerw**
* Fix sending incorrect date values on world change
* Fall particles: check for valid Y
* Fix block break particles
* Streamline player abilities handling
* Update entity sizes
* Fix incorrect name in deserialiser for zombie pigmen
* Remove unused GetClassStatic in cWorld
* Unify multiprotocol entity animations
* Improve bed handling robustness
* Fix thrown snowball/egg hit animations
* Add magical critical effect
* Add animations for shield/item block & break
* More cProtocol cleanup
* cPlayer: move some constants out of header
* Thrown potions/enderpearls: cleanup
* Hoppers: use 'locked' bit in meta
* Implement random ticks more faithfully

### CODE REVIEW

**Comment by 12xx12** [comment]
On it

File: src/WorldStorage/NamespaceSerializer.cpp [hunk 1]
```diff
@@ -501,7 +501,7 @@ eMonsterType NamespaceSerializer::ToMonsterType(const std::string_view a_ID)
 		{ "zoglin",           mtZoglin },
 		{ "zombie",           mtZombie },
 		{ "zombie_horse",     mtZombieHorse },
-		{ "zombie_pigman",    mtZombiePigman },
+		{ "zombified_piglin", mtZombiePigman },
```
Comment (12xx12) [inline src/WorldStorage/NamespaceSerializer.cpp:504]: this was intended to provide support for old save files. the other version is further up/down

File: src/Protocol/Protocol_1_9.cpp [hunk 2]
```diff
@@ -51,9 +51,11 @@ Implements the 1.9 protocol classes:
...
 
 
-/** Value for main hand in Hand parameter for Protocol 1.9. */
-static const UInt32 MAIN_HAND = 0;
-static const UInt32 OFF_HAND = 1;
+// Value for main hand in Hand parameter for Protocol 1.9.
+#define MAIN_HAND 0
+
+// Value for left hand in MainHand parameter for Protocol 1.9.
+#define LEFT_HAND 0
```
Comment (12xx12) [inline src/Protocol/Protocol_1_9.cpp:58]: Is this intended to be the same value?

File: src/WorldStorage/NamespaceSerializer.cpp (hunk 1 above)
Comment (tigerw) [reply src/WorldStorage/NamespaceSerializer.cpp:504]: I see a "PigZombie" mapping. Were we saving as "zombie_pigman" before?

File: src/Protocol/Protocol_1_9.cpp (hunk 2 above)
Comment (tigerw) [reply src/Protocol/Protocol_1_9.cpp:58]: Yeah, the first is for main/offhand attack, second is for the left handed player/right handed player setting.

File: src/WorldStorage/NamespaceSerializer.cpp (hunk 1 above)
Comment (12xx12) [reply src/WorldStorage/NamespaceSerializer.cpp:504]: Just in case someone tried to load a map from a version that uses the name

Comment (12xx12) [reply src/WorldStorage/NamespaceSerializer.cpp:504]: So I'd suggest keeping both

Comment (tigerw) [reply src/WorldStorage/NamespaceSerializer.cpp:504]: Done

**Review by 12xx12 — APPROVED** [review]

**Comment by 12xx12** [comment]
I only saw the zombie pigman change in the last update - is that right?

**Comment by tigerw** [comment]
Many thanks, yes, only changed NamespaceSerializer.cpp.

**Pull request merged**